hudi-common/src/main/java/org/apache/hudi/common/model/HoodieAvroRecord.java

```diff
@@ -126,12 +126,16 @@ public HoodieRecord joinWith(HoodieRecord other, Schema targetSchema) {
   }
 
   @Override
-  public HoodieRecord prependMetaFields(Schema recordSchema, Schema targetSchema, MetadataValues metadataValues, Properties props) {
+  public Option<HoodieRecord> prependMetaFields(Schema recordSchema, Schema targetSchema, MetadataValues metadataValues, Properties props) {
     try {
       Option<IndexedRecord> avroRecordOpt = getData().getInsertValue(recordSchema, props);
-      GenericRecord newAvroRecord = HoodieAvroUtils.rewriteRecordWithNewSchema(avroRecordOpt.get(), targetSchema);
-      updateMetadataValuesInternal(newAvroRecord, metadataValues);
-      return new HoodieAvroRecord<>(getKey(), new RewriteAvroPayload(newAvroRecord), getOperation(), this.currentLocation, this.newLocation);
+      if (avroRecordOpt.isPresent()) {
```

Comment:
so, this is the main fix in this patch is it ?

Reply:
can you help me understand which flow/use-case will hit this. From what I gleaned, most of the callers to this method already filter for deleted record and should not be calling this.